src/arch/ArchHooks/ArchHooks_Win32Static.cpp

```diff
@@ -8,44 +10,37 @@
 
 #include <cstdint>
 #include <vector>
-
-// for QueryPerformanceCounter
 #include <windows.h>
 #include <mmsystem.h>
+
 #if defined(_MSC_VER)
 #pragma comment(lib, "winmm.lib")
 #endif
 
-static bool g_bTimerInitialized;
+constexpr std::int64_t MICROSECONDS_IN_SECOND = 1000000;
+
+static bool was_timer_initialized = false;
 
 /* QueryPerformanceCounter variables below.
  * QueryPerformanceCounter and QueryPerformanceFrequency expect
  * a LARGE_INTEGER, which is a union. These functions store data
  * in the QuadPart of the LARGE_INTEGER, which is a 64-bit integer. */
-LARGE_INTEGER g_liFrequency, g_liStartTime, g_liCurrentTime;
+static LARGE_INTEGER ticks_per_second;
+LARGE_INTEGER current_time;
 
 static void InitTimer()
 {
-	if( g_bTimerInitialized )
-		return;
-	g_bTimerInitialized = true;
-
-	// Grab important information for QPC during the timer initialization
-	QueryPerformanceCounter(&g_liStartTime);
-	QueryPerformanceFrequency(&g_liFrequency);
+	if (!was_timer_initialized) {
+		was_timer_initialized = true;
+		QueryPerformanceFrequency(&ticks_per_second);
+	}
 }
 
 std::int64_t ArchHooks::GetMicrosecondsSinceStart(bool bAccurate)
 {
-	// Make sure the timer is initialized
-	if (!g_bTimerInitialized)
-		InitTimer();
-
-	// Get the current time
-	QueryPerformanceCounter(&g_liCurrentTime);
-
-	// Calculate the elapsed time in microseconds.
-	return ((g_liCurrentTime.QuadPart - g_liStartTime.QuadPart) * 1000000.0) / g_liFrequency.QuadPart;
+	static bool initialized = (InitTimer(), true);
```

Comment:
The notation is as suggested is defined here in Bjarne Stroustrup's C++ third edition, or here on cppreference.com: https://en.cppreference.com/w/cpp/language/operator_other#Comma_operator

> The operators , (comma), && (logical and), and I | (logical or) guarantee that their left-hand operand is evaluated before their right-hand operand. For example, b= (a=2,a+1) assigns 3 to b Examples of the use of and && can be found in $6,2.3. For built-in types, the second operand of && is evaluated only if its first operand is true, and the second operand of I is evaluated only if its first operand is false; this is sometimes called short-circuit evaluation. Note that the sequencing operator , (comma) is logically different from the comma used to separate arguments in a function call. Consider:
> ```
>  fl (v[i], i++); f2( (v[i],i++) );
>  // two arguments 1/ one argument
> ```
>  The call of f1 has two arguments, v [i] and it+, and the order of evaluation of the argument expressions is undefined. Order dependence of argument expressions is very poor style and has undefined behavior. The call of f2 has one argument, the comma expression (v[il,i++), which is equivalent to i++

(source: https://github.com/user-attachments/assets/2adb0bdc-9df3-44bd-b3f1-dd9b510126de)

I chose it because I thought it looks nicer to have it all on one line. It also makes it clear that once `InitTimer()` is called, the boolean value is permanently set to `true`.

The reason for making it a `static bool` is because a static bool can only be evaluated once. This guarantees no other thread has its own idea of when the game started or something like that. It won't be evaluated a second time, so once the timer is initialized, it simply returns true.

The name `initialized` is meaningless, it could be anything.

Reply:
I would just do two statements. This notation is pretty esoteric.

Also my question was why do we have two variables that seemingly mean the same thing.

Reply:
> I would just do two statements. This notation is pretty esoteric.
> 
> Also my question was why do we have two variables that seemingly mean the same thing.

I see what you meant now. Got it